### DISCUSSION TITLE
How can you use the `colorconvert` functions when the source image has an ICC profile?

### DISCUSSION THREAD

**Discussion by MarkCallow** [OpenImageIO/oiio#3114] [Q&A]
I've been studying the OIIO documentation looking for the answer to this and so far haven't seen it. If I have an source image such as a PNG image with an ICC profile, which the PNG reader provides as a metadata item, how do I use the colorconvert functions to transform from that profile to some other color space, e.g. sRGB.

**Comment by lgritz** [comment]
That's not currently implemented, but I think (?) that there is functionality in OCIOv2 that would support this? So I think that with a little work, we could make this work properly when built against OCIOv2.

**Comment by Upliner** [comment]
Any updates on this?

**Reply by lgritz** [reply]
I haven't had time to do any work on this yet. I think most of the ingredients are all there -- we can retrieve the ICC profile from an image file, and we have a class that wraps OCIO ColorConfig (to which we could add a method that will do a transform based on the ICC, assuming that OCIO has such a call now). Just need to mostly connect the pieces.

**Comment by NayamAmarshe** [comment]
Any updates?

**Comment by lgritz** [comment]
No update, I'm sorry to report. I'm afraid we are waiting for somebody who knows more than me about ICC profiles (and maybe about OpenColorIO APIs) to come forward to do this work. 

As we say on this project: 🎣 Mackerel!

**Comment by lgritz** [comment]
Does anybody with deeper OpenColorIO knowledge than me know the answer to whether this is possible? Does OCIO have APIs that let you, I guess, invert the display transform implied by the embedded ICC profile to transform an image into one of our canonical spaces?

**Reply by doug-walker** [reply]
This is definitely possible. Basically the steps would be:
1. Save the embedded ICC profile as a separate file.
2. Load an OCIO built-in config and add a new display color space to it, using a FileTransform pointing to the ICC profile.
3. Use a ColorSpaceTransform or DisplayViewTransform to convert to one of the other color spaces in the built-in config.

I'm too swamped at the moment to make a PR, but would be willing to advise someone else that wants to pick up the Mackerel challenge.
